spring-boot-actuator/src/main/java/org/springframework/boot/actuate/autoconfigure/HealthIndicatorAutoConfiguration.java

```diff
@@ -363,4 +392,51 @@ protected ElasticsearchHealthIndicator createHealthIndicator(Client client) {
 
 	}
 
+	/**
+	 * A {@link BeanPostProcessor} that registers {@link HealthIndicator} beans with a
+	 * {@link HealthIndicatorRegistry}.
+	 */
+	private static class HealthIndicatorRegistryPostProcessor implements BeanPostProcessor {
```

Comment:
Also, if I don't want all of the HealthIndicator beans pushed into my custom registry, the post processor will do so anyway. If a custom registry is provided, it needs to be possible to have complete control over its contents.